### ISSUE TITLE
Feature request, boolean property shorthand

### ISSUE THREAD

**Issue by rictic** [Polymer/lit-html#989]
Passing this along from Rubén López. It would be useful to be able to set boolean properties to true with some shorthand:

```typescript
// this
html`
  <my-elem .bool></my-elem>
`

// would behave like
html`
  <my-elem .bool=${true}></my-elem>
`;
```

I'm not entirely sold on it, as it adds complexity to the parser and the user's mental model, but with lit-analyzer around we can make it an error to use this syntax with unknown or non-boolean properties.

**Comment by CaptainCodeman** [comment]
Why wouldn't this be used?
```html
<my-elem bool></my-elem>
```

**Comment by ruphin** [comment]
This is not possible. We don't parse the contents of templates, except those surrounding escaped values to determine what part to create. (more or less)

The feature request was about setting properties, not attributes. @CaptainCodeman

**Comment by CaptainCodeman** [comment]
Yes, but the attribute normally sets the boolean property (when using lit-element).

**Comment by ruphin** [comment]
This is about the templating system. `<my-elem>` in this example is not necessarily LitElement based. It may be any type of Element, that may not set the property based on presence of attributes.

In addition to that, setting attributes is generally more expensive than properties.

**Comment by rubenlg** [comment]
@ruphin when you say "not possible" do you mean there is some fundamental limitation completely outside of the control of lit-html code, or do you mean it's not possible without significant changes to the internals of lit-html?

To clarify where this is coming from, I'm trying to migrate Polymer templates to lit-html. In Polymer, code looking like regular HTML markup (without extra symbols around attributes) actually sets properties, not attributes, and we have a bunch of code that looks like this: 
```html
<my-elem bool></my-elem>
```

Migrating that code to lit-html makes it harder to read for humans: 
```html
<my-elem .bool=${true}></my-elem>
```

That's why I was suggesting having a shorthand syntax to reduce the boilerplate and make it easier to read:
```html
<my-elem .bool></my-elem>
```

Without changes to lit-html, there is a trade-off between convenience and performance. If we want convenience, we can use the attribute syntax and rely on serialization, and if we want performance, we should use the uglier syntax, right?

It's entirely possible that this is happening because we are not yet used to reading lit-html code, and that way of setting static boolean attributes taxes a bit more our brains. Perhaps this is temporary?

**Comment by csvn** [comment]
@rubenlg I think by "not possible" it means the performance impact of parsing all attributes on elements (event those that are "constant"). I'ts probably much more preferred and better if the custom element base class (e.g. `LitElement` or your own) handles the attribute:

```js
import { customElement, LitElement } from 'lit-element';

// Note: There is also a non-decorator syntax
@customElement('my-elem')
export class MyElem extends LitElement {
  @property({ type : Boolean, reflect: true }) bool = false;
}
```

This way, only the components that need to transform attributes to property value (`<my-elem bool> => myElem.bool = true`) will have any performance impact.

**Comment by rubenlg** [comment]
Makes sense. @rictic, feel free to close this issue. The thread should serve as documentation for others going through the same migration.
